modules/programs/comodoro.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (config.programs) comodoro;
+  tomlFormat = pkgs.formats.toml { };
+
+  server = config.services.comodoro;
+  serverArgs = with server.settings; {
+    inherit preset;
+    protocols = if lib.isList protocols then
+      lib.concatStringsSep " " protocols
+    else
+      protocols;
+  };
+
+in {
+  meta.maintainers = with lib.hm.maintainers; [ soywod ];
+
+  options = {
+    programs.comodoro = {
+      enable =
+        lib.mkEnableOption "Enable Comodoro, the CLI to manage your time.";
+      package = lib.mkPackageOption pkgs "comodoro" { };
+      settings = lib.mkOption {
+        type = lib.types.submodule { freeformType = tomlFormat.type; };
+        default = { };
+        description = ''
+          Comodoro configuration.
+          See <link xlink:href="https://pimalaya.org/comodoro/cli/configuration/"/> for supported values.
+        '';
+      };
+    };
+
+    services.comodoro = {
+      enable = lib.mkEnableOption "Enable the Comodoro server.";
+
+      environment = lib.mkOption {
+        type = with lib.types; attrsOf str;
+        default = { };
+        example = lib.literalExpression ''
+          {
+            "PASSWORD_STORE_DIR" = "~/.password-store";
+          }
+        '';
+        description = ''
+          Extra environment variables to be exported in the service.
+        '';
+      };
+
+      settings = {
+        preset = lib.mkOption {
+          type = lib.types.nonEmptyStr;
+          description = ''
+            Use configuration from the given preset as defined in the configuration file.
+          '';
+        };
+
+        protocols = lib.mkOption {
+          type = with lib.types; nonEmptyListOf nonEmptyStr;
+          description = ''
+            Define protocols the server should use to accept requests.
+          '';
+        };
```

Comment:
Are there any good defaults for this or will users have to specify both of these values to run the server?

Reply:
These values are mandatory ones, hence the `nonEmpty*`.